### PR TITLE
feat(core/a11y): add axe-core integration

### PR DESCRIPTION
**Pull request by sidvishnoi**
Closes https://github.com/w3c/respec/issues/2777

- The linter API is not flexible enough with error reporting, so created a new core/a11y module.
- Axe is quite slow on W3C specs. Need to disable a lot of rules, or provide authors a way to select rules. Should be used locally only to fix issues, not on live versions.

### CODE REVIEW

File: src/core/a11y.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+const DISABLED_RULES = [
+  "landmark-one-main", // need to add a <main>, else it marks entire page as errored
+  "landmark-unique",
+  "region",
+  "color-contrast", // too slow 🐢
+];
+
+export async function run(_conf) {
+  // TODO: run conditionally based on conf
+  const violations = await getViolations();
```
Comment (marcoscaceres) [inline src/core/a11y.js]: Try/catch, as we discussed on Slack.

File: src/core/a11y.js [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    elementRef: true,
+    resultTypes: ["violations"],
+    reporter: "v1", // v1 includes a `failureSummary`
+  };
+
+  let axe;
+  try {
+    axe = await importAxe();
+  } catch (error) {
+    pub("error", "Failed to load a11y linter.");
```
Comment (marcoscaceres) [inline src/core/a11y.js]:
```suggestion
    pub("error", "Failed to load a11y linter. See developer console for details.");
```

Comment (sidvishnoi) [reply src/core/a11y.js]: Heh. I added it and then removed it for no reason.

**Comment by sidvishnoi** [comment]
Demo: [Manifest spec with color-contrast rule](https://respec-preview.netlify.com/?spec=https%3A%2F%2Fw3c.github.io%2Fmanifest%2F&version=https%3A%2F%2Fdeploy-preview-2779--respec-pr.netlify.com%2Frespec-w3c.js&respecConfig=respecConfig.a11y+%3D+%7B+runOnly%3A+%5B%27color-contrast%27%5D+%7D)
Warning: color-contrast rule is slow and will hand the document. Shown in example to demonstrate rules can be overridden if needed.

File: src/core/a11y.js [hunk 3]
```diff
@@ -0,0 +1,103 @@
+import { pub } from "./pubsubhub.js";
```
Comment (marcoscaceres) [inline src/core/a11y.js]: Might be nice to have a short pre-amble just stating what the module does and how it works.

File: src/core/a11y.js [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+
+function importAxe() {
+  return new Promise((resolve, reject) => {
+    const script = document.createElement("script");
+    script.classList.add("remove");
+    script.src = "https://unpkg.com/axe-core@3/axe.min.js";
+    script.onload = () => resolve(window.axe);
+    script.onerror = reject;
+    document.head.appendChild(script);
+  });
```
Comment (marcoscaceres) [inline src/core/a11y.js]: Nit: 

```suggestion
  const script = document.createElement("script");
  script.classList.add("remove");
  script.src = "https://unpkg.com/axe-core@3/axe.min.js";
  document.head.appendChild(script);
  return new Promise((resolve, reject) => {
    script.onload = () => resolve(window.axe);
    script.onerror = reject;
  });
```

File: src/core/a11y.js [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+}
+
+/** @returns {Promise<typeof window.axe>} */
+function importAxe() {
+  const script = document.createElement("script");
+  script.classList.add("remove");
+  script.src = "https://unpkg.com/axe-core@3/axe.min.js";
+  document.head.appendChild(script);
+  return new Promise((resolve, reject) => {
+    script.onload = () => resolve(window.axe);
```
Comment (saschanaz) [inline src/core/a11y.js]: So it works as a global script? Bit sad...

Comment (sidvishnoi) [reply src/core/a11y.js]: :crying_cat_face:
They do plan to make it a ES module though.

**Comment by marcoscaceres** [comment]
I can write the docs if you’d like.

File: src/core/a11y.js (hunk 5 above)
Comment (saschanaz) [reply src/core/a11y.js]: Maybe rebuild as a module, as we do for jQuery and pluralize? Bit different as they are already CJS but should still be doable.

Comment (sidvishnoi) [reply src/core/a11y.js]: That would mean we either bundle this ~400KB library into ReSpec, or ask W3C to host it separately. unpkg seems fine to me, given how infrequently this feature would be used (won't be used in live versions for sure, as it's slow... but only during local spec authoring)

**Comment by sidvishnoi** [comment]
I wrote a draft at https://github.com/w3c/respec/wiki/a11y

**Comment by marcoscaceres** [comment]
Thanks for updating the docs! I did a little cleanup. I think the expectation is run Axe on CI, but also on normal specs. I'll let the a11y folks know about the potential pitfalls of enabling too many features - or that they should only enable certain features in CI.

**Pull request merged**